tests/test_async_filelock.py

```diff
@@ -0,0 +1,181 @@
+from __future__ import annotations
+
+import logging
+from pathlib import Path, PurePath
+
+import pytest
+
+from filelock import AsyncFileLock, AsyncSoftFileLock, BaseAsyncFileLock, Timeout
+
+
+@pytest.mark.parametrize("lock_type", [AsyncFileLock, AsyncSoftFileLock])
+@pytest.mark.parametrize("path_type", [str, PurePath, Path])
+@pytest.mark.parametrize("filename", ["a", "new/b", "new2/new3/c"])
+@pytest.mark.asyncio()
+async def test_simple(
+    lock_type: type[BaseAsyncFileLock],
+    path_type: type[str | Path],
+    filename: str,
+    tmp_path: Path,
+    caplog: pytest.LogCaptureFixture,
+) -> None:
+    caplog.set_level(logging.DEBUG)
+
+    # test lock creation by passing a `str`
+    lock_path = tmp_path / filename
+    lock = lock_type(path_type(lock_path))
+    async with lock as locked:
+        assert lock.is_locked
+        assert lock is locked
+    assert not lock.is_locked
+
+    assert caplog.messages == [
+        f"Attempting to acquire lock {id(lock)} on {lock_path}",
+        f"Lock {id(lock)} acquired on {lock_path}",
+        f"Attempting to release lock {id(lock)} on {lock_path}",
+        f"Lock {id(lock)} released on {lock_path}",
+    ]
+    assert [r.levelno for r in caplog.records] == [logging.DEBUG, logging.DEBUG, logging.DEBUG, logging.DEBUG]
+    assert [r.name for r in caplog.records] == ["filelock", "filelock", "filelock", "filelock"]
+    assert logging.getLogger("filelock").level == logging.NOTSET
+
+
+@pytest.mark.parametrize("lock_type", [AsyncFileLock, AsyncSoftFileLock])
+@pytest.mark.parametrize("path_type", [str, PurePath, Path])
+@pytest.mark.parametrize("filename", ["a", "new/b", "new2/new3/c"])
+@pytest.mark.asyncio()
+async def test_acquire(
+    lock_type: type[BaseAsyncFileLock],
+    path_type: type[str | Path],
+    filename: str,
+    tmp_path: Path,
+    caplog: pytest.LogCaptureFixture,
+) -> None:
+    caplog.set_level(logging.DEBUG)
+
+    # test lock creation by passing a `str`
+    lock_path = tmp_path / filename
+    lock = lock_type(path_type(lock_path))
+    async with await lock.acquire() as locked:
+        assert lock.is_locked
+        assert lock is locked
+    assert not lock.is_locked
+
+    assert caplog.messages == [
+        f"Attempting to acquire lock {id(lock)} on {lock_path}",
+        f"Lock {id(lock)} acquired on {lock_path}",
+        f"Attempting to release lock {id(lock)} on {lock_path}",
+        f"Lock {id(lock)} released on {lock_path}",
+    ]
+    assert [r.levelno for r in caplog.records] == [logging.DEBUG, logging.DEBUG, logging.DEBUG, logging.DEBUG]
+    assert [r.name for r in caplog.records] == ["filelock", "filelock", "filelock", "filelock"]
+    assert logging.getLogger("filelock").level == logging.NOTSET
+
+
+@pytest.mark.parametrize("lock_type", [AsyncFileLock, AsyncSoftFileLock])
+@pytest.mark.asyncio()
```

Comment:
Let us split this up so that one that is only testing one thing.

Reply:
This doesn't seem to be easy to split. Is there a more detailed split plan? In fact, it is basically consistent with the unit test of synchronized file locks.